test_opensearchpy/test_async/test_signer.py

```diff
@@ -18,19 +18,38 @@
 
 
 class TestAsyncSigner:
-    def mock_session(self):
+    def mock_session(self, disable_get_frozen=True):
         access_key = uuid.uuid4().hex
         secret_key = uuid.uuid4().hex
         token = uuid.uuid4().hex
         dummy_session = Mock()
         dummy_session.access_key = access_key
         dummy_session.secret_key = secret_key
         dummy_session.token = token
+        dummy_session.get_frozen_credentials = Mock(return_value=dummy_session)
+
+        if disable_get_frozen:
```

Comment:
There's probably a cleaner way to do this, if you care, by inheriting `TestAsyncSigner` into a `TestAsyncSignerWithFrozenCredentials(TestAsyncSigner)` or something like that?

Reply:
Hrm. The issue is the checks for the `get_frozen_credentials` method in the signer implementations will always evaluate to `True` if I don't explicitly delete the attribute because of how eager `Mock` is to mock :)